book/course/theory.md

```diff
@@ -0,0 +1,195 @@
+```
+
+Executable installers, may not include dependencies (other files required to run); these could be provided by default by the Operating System or may require installing separately.
+
+### Package Managers
+
+Package managers are used to manage applications, tools and libraries as packages.
+
+Packages are archives (often compressed files like zip) and can contain any combination of software, libraries and their metadata.
+
+Packages are collected and maintained together as distributions (eg conda-forge).
+
+Package managers keep track of packages and all of their versions/variants, both installed and available to be installed, along with their dependencies.
+
+They usually allow the following functionality:
+
+* Searching (for both online repositories and installed)
+* Downloading from package repositories.
+* installation and removal of packages.
+* Upgrading/downgrading package versions.
+* Installation and removal of dependancies.
+
+The most commonly used package managers on ARC3/ARC4 are anaconda and pip.
+
+* [Anaconda](https://arcdocs.leeds.ac.uk/software/compilers/anaconda.html) is an environment and package manager, primarily for Python and R.
+* [pip](https://en.wikipedia.org/wiki/Pip_(package_manager)) is a package manager for Python.
+
+## Build tools
+
+Build tools is a generic term for any programs and tools used to build software. Building software can include steps such as configuration, compiling, testing, packaging.
+
+Common buildtools :
+
+* Autotools - is a set of build tools to help make portal Unix-like software.
+* CMake - is a cross platform build automation tool, to support the creation of configurations for build tools.
+* Make - the traditional Unix/Linux build tool.
+
+### Compilers
+
+A compiler is a program which translates source code written in a programming language into target code. This is typically into machine code to create an executable program.
+
+ARC3 and ARC4 have the [Intel](https://arcdocs.leeds.ac.uk/software/compilers/intel.html) compiler set as default, with the [GNU](https://arcdocs.leeds.ac.uk/software/compilers/gnu.html) and [LLVM](https://arcdocs.leeds.ac.uk/software/compilers/llvm.html) compilers being popular alternatives available on the module system.
+
+#### Interpreters
+
+An interpreter reads and executes source code directly. They generally runs code slower than a compiled executable as it has to compile code every time the code is run, but there are optimisations such as bytecode compilation and caching that can vastly improve performance.
+
+Python is a popular interpreted language and the most common interpreter is the default, [CPython](https://en.wikipedia.org/wiki/CPython).
+
+Some languages are capable of being compiled or run through an interpreter.
+
+### Build Configuration
+
+Tools like CMake can be used to create different platform specific for different build tools configurations.
```

Comment:
```suggestion
Tools like CMake can be used to create different platform specific for different build tools configurations.
```
This doesn't make sense, think you're missing a word after "different platform specific"